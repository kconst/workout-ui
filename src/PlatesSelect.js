import React, { Component, PropTypes } from 'react'
import classnames from 'classnames'
import BarbellWeights from './BarbellWeights.js';

class PlatesSelect extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            weight: this.props.useImperial ? 45 : 20
        }
    }

    componentWillMount() {
        this.setState({
           plates: this.props.useImperial ? [2.5, 5, 10, 25, 45] : [0.5, 1, 2.5, 5, 10, 15, 20, 25]
        });
    }

    increaseWeight(weight) {
       this.setState({
           weight: this.state.weight += weight
       });
    }

    render() {
        return (
            <section className="PlatesSelect">
                <h3>{ this.state.weight }</h3>
                <BarbellWeights useImperial={ false } weight={ this.state.weight }/>
                <ul className="plate-list">
                    {this.state.plates.map((plate, i) =>
                        <li key={i} className={ 'plate-' + (plate + '').replace('.', 'p') } onClick={ () => this.setState({ weight: this.state.weight += (plate * 2) }) }>{ plate }</li>
                    )}
                </ul>
            </section>
        );
    }
}

PlatesSelect.propTypes = {
    
};

export default PlatesSelect
