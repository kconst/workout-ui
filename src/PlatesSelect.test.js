import React from 'react';
import PlatesSelect from './PlatesSelect.js';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

// fixes enzyme's issue with jest
jest.mock('react/lib/ReactDefaultInjection');

it('renders correctly in default state', () => {
  const component = renderer.create(<PlatesSelect useImperial={ false }/>);
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it ('should have the correct total for metric weights', () => {
  const component = shallow(<PlatesSelect useImperial={ false }/>);
  
  component.find('.plate-15').simulate('click');
  
  expect(component.find('h3').text()).toEqual('50');
});

it ('should have the correct total for imperial weights', () => {
  const component = shallow(<PlatesSelect useImperial={ true }/>);
  
  component.find('.plate-45').simulate('click');
  
  expect(component.find('h3').text()).toEqual('135');
});
