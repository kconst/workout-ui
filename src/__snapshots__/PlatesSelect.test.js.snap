exports[`test renders correctly in default state 1`] = `
<section
  className="PlatesSelect">
  <h3>
    20
  </h3>
  <ul
    className="plate-list" />
  <ul
    className="plate-list">
    <li
      className="plate-0p5"
      onClick={[Function onClick]}>
      0.5
    </li>
    <li
      className="plate-1"
      onClick={[Function onClick]}>
      1
    </li>
    <li
      className="plate-2p5"
      onClick={[Function onClick]}>
      2.5
    </li>
    <li
      className="plate-5"
      onClick={[Function onClick]}>
      5
    </li>
    <li
      className="plate-10"
      onClick={[Function onClick]}>
      10
    </li>
    <li
      className="plate-15"
      onClick={[Function onClick]}>
      15
    </li>
    <li
      className="plate-20"
      onClick={[Function onClick]}>
      20
    </li>
    <li
      className="plate-25"
      onClick={[Function onClick]}>
      25
    </li>
  </ul>
</section>
`;
